jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/DoSHandler.java

```diff
@@ -0,0 +1,622 @@
+            {
+                _sampleStart = NanoTime.now();
+            }
+
+            @Override
+            public boolean isRateExceededBySample(long now)
+            {
+                // Count the request
+                _sampleCount++;
+
+                long elapsedTime = NanoTime.elapsed(_sampleStart, now);
+
+                // We calculate the moving average if:
+                //    + the sample exceeds the rate
+                //    + the sample period has been exceeded
+                if (_sampleCount > _maxRequestsPerSecond || (_sampleStart != 0 && elapsedTime > _samplePeriod.toNanos()))
+                {
+                    calculateMovingAverage(now);
+                }
+
+                // if the rate has been exceeded?
+                return _exponentialMovingAverage > _maxRequestsPerSecond;
+            }
+
+            @Override
+            public boolean isIdle(long now)
+            {
+                calculateMovingAverage(now);
+                return _exponentialMovingAverage <= 0.0001;
+            }
+
+            private void calculateMovingAverage(long now)
+            {
+                double elapsedTime1 = (double)(now - _sampleStart);
+                double count = _sampleCount;
+                if (elapsedTime1 > 0.0)
+                {
+                    double currentRate = (count * TimeUnit.SECONDS.toNanos(1L)) / elapsedTime1;
+                    // Adjust alpha based on the ratio of elapsed time to the interval to allow for long and short intervals
+                    double adjustedAlpha = _alpha * (elapsedTime1 / _samplePeriod.toNanos());
+                    if (adjustedAlpha > 1.0)
+                        adjustedAlpha = 1.0; // Ensure adjustedAlpha does not exceed 1.0
+
+                    _exponentialMovingAverage = (adjustedAlpha * currentRate + (1.0 - adjustedAlpha) * _exponentialMovingAverage);
+                }
+                else
+                {
+                    // assume count as the rate for the sample.
+                    double guessedRate = count * TimeUnit.SECONDS.toNanos(1) / _samplePeriod.toNanos();
+                    _exponentialMovingAverage = (_alpha * guessedRate + (1.0 - _alpha) * _exponentialMovingAverage);
+                }
+
+                // restart the sample
+                _sampleStart = now;
+                _sampleCount = 0;
+            }
+
+            double getCurrentRatePerSecond()
+            {
+                return _exponentialMovingAverage;
+            }
+        }
+    }
+
+    /**
+     * A Handler to reject DoS requests with a status code or failure.
+     */
+    public static class StatusRejectHandler implements Request.Handler
+    {
+        private final int _status;
+
+        public StatusRejectHandler()
+        {
+            this(-1);
+        }
+
+        /**
+         * @param status The status used to reject a request, or 0 to fail the request or -1 for a default ({@link HttpStatus#TOO_MANY_REQUESTS_429}.
```

Comment:
Can this javadoc be moved to the class?

Reply:
The class javadoc does say pretty much this?